source/vecs/entitymanager.d

```diff
@@ -22,12 +22,14 @@ version(vecs_unittest)
 	import core.exception : AssertError;
 }
 
+alias EntityManager = EntityManagerImpl!(void delegate() @safe);
 
 /**
  * Responsible for managing all entities lifetime and access to components as
  *     well as any operation related to them.
  */
-class EntityManager
+class EntityManagerImpl(Fun)
+	if(is(Fun : void delegate()))
```

Comment:
No, it does not. This is used to get the attributes of the `delegate`. Signal only works with `delegate` internally, so passing a `function` could lead to confusion. Signal accepts function pointers and converts them to delegates, but **only** when using `connect`. The type definition, be it by the template `attributes` or template `parameters`, must always be a `delegate`.

Reply:
Yes, but AFAIK this is only to extract the function attributes. There's no need for such enforcement

Reply:
But for an outside user, or a user using the lib for the first time it might get confusing because he is not only passing the attributes, he is passing a type. If I make this too loose, I think it'll be confusing. *Does my Signal only work with function now?*, *I passed a function, how can I use delegates?*. Not only questions like these, but also the lib usage would not be consistent. Some users would use `function` others `delegate` and in the worst case, you could even use a function by passing its pointer as the type.
```d
void fn() {}

new EntityManagerImpl!(void function());
new EntityManagerImpl!(void delegate());
new EntityManagerImpl!(&fn);
```

I don't think that restricting the user to only `delegate` is a bad thing. It brings consistency, clarity and it is conformant with how Signal works.

Reply:
A `function` is convertible to a `delegate`. But sure, I agree with your point. Consistency is preferable and leads to less confusion. :+1: